File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s):
        d = {}
        seen = set()
        for idx, c in enumerate(s):
            if c not in seen:
                d[c] = idx
                seen.add(c)
            elif c in d:
                del d[c]
        return min(d.values()) if d else -1

File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.java
class Solution {
public:
    int firstUniqChar(string s) {
        if(s.length()==1) return 0;
        unordered_map<char,int> mp;
        for(auto i:s) mp[i]++;
        for(int i=0;i<s.size();++i)
            if(mp[s[i]]==1)
                return i; 
          return -1;  
     
    }
};